### DISCUSSION TITLE
Branding Solution, FG's Way

### DISCUSSION THREAD

**Discussion by fredgrott** [rydmike/flex_color_scheme#195] [Ideas]
So everyone the MD3 spec doc is still problematic in explaining this so this is my approach:

1. On Android, we start with a MD3 baseline color scheme.
2. Then the Android system computes the dominant color in wallpaper and creates a new color scheme that shifts the hues in the MD3 baseline to generate the new color scheme and that is the corePalette that the dynamic color package grabs to compute the dynamic color schemes for flutter usage.

While we do not have access to the wallpaper colors down to the android level, we can still approximate it so that we come out correct for branding as we at maximum can only have two color schemes at once on the screen, i.e. user generated color scheme and content generated color scheme. We can get that accomplished by creating a custom dynamic color builder that hue shifts the dynamic color scheme it generates with our own brand baseline color scheme.

I am going to do a full separate github repo showing the approach which will be up in a week or two.

The inspiration was how freaking horrible the Material Design 3 Theme Builder is, as I somewhat prefer the superior Flex Color Scheme playground!

Side benefit is the approach will still work when the new surface tones and fixed tones hit the ColorScheme class.

**Comment by rydmike** [comment]
Ok cool, sounds interesting 😄 👍🏻
